packages/cli/src/lib/snapshot/snapshotReporter.ts

```diff
@@ -57,6 +58,60 @@ export class SnapshotReporter {
     );
   }
 
+  private reportHandlers: Record<
+    SnapshotReportStatus,
+    (this: SnapshotReporter) => void | Promise<void>
+  > = {
+    [SnapshotReportStatus.SUCCESS]: () => {},
+    [SnapshotReportStatus.MISSING_VAULT_ENTRIES]: () => {},
+    [SnapshotReportStatus.ERROR]: () => {},
+  };
+
+  public setReportHandler(
+    status: SnapshotReportStatus,
+    handler: (this: SnapshotReporter) => void | Promise<void>
+  ): SnapshotReporter {
+    this.reportHandlers[status] = handler;
+    return this;
+  }
+
+  public async handleReport(): Promise<void> {
+    await this.reportHandlers[this.getReportStatus()].apply(this);
+  }
+
+  public getReportStatus(): SnapshotReportStatus {
+    if (this.isSuccessReport()) {
+      return SnapshotReportStatus.SUCCESS;
+    }
+    if (this.isVaultEntriesMissingReport()) {
+      return SnapshotReportStatus.MISSING_VAULT_ENTRIES;
+    }
+    return SnapshotReportStatus.ERROR;
+  }
+
+  private isVaultEntriesMissingReport(): boolean {
+    for (const resourceTypeEntry of Object.values(
+      this.report.resourceOperationResults
+    )) {
+      for (const errors of Object.values(resourceTypeEntry)) {
+        for (const err of errors) {
+          if (!SnapshotReporter.isVaultEntryMessage(err)) {
+            return false;
+          }
+        }
+      }
+    }
+    return true;
+  }
+
+  private static isVaultEntryMessage(err: string): boolean {
+    // TODO CDX-939: Define contract with backend for report and upcoming contract.
+    // Current 'contract' 😅:
+    return /^The vault entry referenced by.*could not be found in the vault\.$/.test(
+      err
+    );
+  }
+
```

Comment:
Why do you need to validate that via a regex?
There is already a call for that
https://github.com/coveo/platform-client/blob/master/src/resources/Vaults/Vaults.ts#L16

Reply:
We discussed it on meet, so I'll summarize:
We saw that on the [Migration Service]( https://github.com/coveo/migrationservice/blob/1d3ac8207282942e1828f942d3858ea1f36baa81/coveo-cloud-services-migration/src/main/java/com/coveo/cloud/migration/service/interpolation/SnapshotPlaceholderInterpolator.java#L95-L128)
The errors are aggregated.

That means that if there's more than one error on a resource we'll get them all in one go, allowing us to offer a comprehensive report to the user.

This will avoid the number of 'attempts' a user would do before successfully running its push (assuming he digests the report properly, but if he doesn't that'll be on our end to fix that).

I remember my first days with the Coveo For Salesforce package, when the MDAPI was the only way to push (and therefore test) stuff on an SDO, it was a hell circus circle: Launch a deploy, wait 10', get 1 error, fix it, repeat. :feelsgood: 

I would really not want to impose that on our users.